### ISSUE TITLE
Remove inline javascript

### ISSUE THREAD

**Issue by nijel** [phpmyadmin/phpmyadmin#12261]
There are currently several places in our code which use inline javascript (mostly `onclick`, `onsubmit` and `onchange`). These should be removed and placed into javascript files.

After removal we can get rid of `'unsafe-inline'` for scripting in CSP.

**Comment by gavinkalika** [comment]
Is there a way to claim this issue? I would like to work on it.

**Comment by nijel** [comment]
Commenting here is enough :-).

**Comment by architkansal** [comment]
@gavinkalika if you are not working on it, I would like to take that bug.

**Comment by gavinkalika** [comment]
go for it

**Comment by architkansal** [comment]
@nijel which all php files I should edit and do I create separate files in the same directories or edit some existing js files?

**Comment by nijel** [comment]
Probably it makes sense to add the code to existing files. Most of that will be binding of existing callbacks to elements.

Also it's not only about PHP files, we do generate HTML with inline javascript as well (for example see [js/pmd/history.js](https://github.com/phpmyadmin/phpmyadmin/blob/master/js/pmd/history.js#L91).

**Comment by eragon512** [comment]
is this issue still open? i would like to work on it, if possible

**Comment by nijel** [comment]
Yes, it's open. AFAIK there are still enough places left.

**Comment by manishbisht** [comment]
I have checked for the places where the `onclick`, `onsubmit` and `onchnage` are used. So we have to replace these inline codes with the javascript files.
![1](https://cloud.githubusercontent.com/assets/14255990/23561279/a1b89f82-0063-11e7-9894-06c18494da1f.png)
![2](https://cloud.githubusercontent.com/assets/14255990/23561280/a1fb1b32-0063-11e7-8495-ad59020b3451.png)
![3](https://cloud.githubusercontent.com/assets/14255990/23561282/a23a3650-0063-11e7-940c-c9c7899c6aee.png)

**Comment by plartem** [comment]
there are also few onmousedown, onmouseover and onmouseout

**Comment by manishbisht** [comment]
These are for `onmousedown`, `onmouseover` and `onmouseout`
![4](https://cloud.githubusercontent.com/assets/14255990/23561732/81d62110-0065-11e7-899c-dfd8bedc9761.png)
![5](https://cloud.githubusercontent.com/assets/14255990/23561733/81d9cf4a-0065-11e7-92fe-587b1eb895c2.png)
![6](https://cloud.githubusercontent.com/assets/14255990/23561734/81e7b970-0065-11e7-867f-fe706a939256.png)

The way this task can be done it by giving some id to it and then using their id javascript can add event listeners to them for onclick, onsubmit, onchange, onmousedown, onmouseover and onmouseout events.

**Comment by manishbisht** [comment]
@nijel @ibennetch I am interested in working on this issue I have found all the places where this issue need changes.

The way this task can be done it by giving some id to it and then using their id javascript can add event listeners to them for onclick, onsubmit, onchange, onmousedown, onmouseover and onmouseout events. I want to where can i put the new javascript should I create a new js file or I can add it on the same file with the `script` tag

**Comment by nijel** [comment]
Feel free to start working on this. You even don't have to submit complete fix, submitting it by pieces is okay as well.

The code should be reviewed and not automatically converted be attached to ID. In some cases, the fix can be different, for example the whole thing can be done by CSS in this case:

```
onmouseover="this.className='small_tab2';" 
onmouseout="this.className='small_tab';"
```

**Comment by manishbisht** [comment]
I think it should be reopened as it is not completely fixed.

**Comment by nijel** [comment]
Good point :-). Please do not include Fixes #... in the commit message in that case :-). You can just reference the issue as "Issue #...".

**Comment by manishbisht** [comment]
Thanks for your suggestion. Coming to onmousedownevent it contains the use of php in javascript events. So the way I think of to implement is to create a new java script file with `.php` or `.phtml` extension to handle the event and then I will put that file inside this folder `https://github.com/phpmyadmin/phpmyadmin/tree/master/templates/javascript` Please correct me if I am voilating the coding standards somewhere made by phpmyadmin or any suggestions for putting the new code inside existing file.

**Comment by manishbisht** [comment]
Current code is this
`<td id="id_zag_<?= htmlspecialchars($t_n_url); ?>"
          class="tab_zag nowrap tab_zag_noquery"
          onmousedown="cur_click=document.getElementById('<?= $t_n_url; ?>');"`
And now I am going to replace it with this code
`$( "#id_zag" ).mousedown(function() {
          cur_click=document.getElementById('<?= $t_n_url; ?>');
});`
To add this a new javascript file will be required with `.php` or `.phtml` because of php in the code.
I think this is what you mean to use jquery library or you want the completely different approach to implement this.

**Comment by nijel** [comment]
Please don't add another not necessary complexity to the code. That element is actually parent of the clicked element, so can be easily figured out from the javascript without knowing it's id.

**Comment by manishbisht** [comment]
Yes, By looking more into codebase I have figured out that way. I have implemented it using that way only. Can you have a look on the referenced PR.

**Comment by nijel** [comment]
Apparently this should not have been closed, but there is now  #13270 covering rest of the issues....
